src/screens/SearchScreen.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import { ScrollView, StyleSheet } from 'react-native';
+
+import EditScreenInfo from 'components/EditScreenInfo';
+import { Text, View } from 'components/Themed';
+import { useEffect, useState } from 'react';
+import { getAllTrees } from 'src/database/firebase';
+import { Tree } from '@types';
+import SearchCard from 'src/components/SearchCard';
+import { Searchbar } from 'react-native-paper';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+  title: {
+    fontSize: 20,
+    fontWeight: 'bold',
+  },
+});
+
+export default function SearchScreen() {
+  const [searchQuery, setSearchQuery] = useState('');
+  const [trees, setTrees] = React.useState<Tree[]>([]);
+  useEffect(() => {
+    async function getTrees() {
+      let data = await getAllTrees();
```

Comment:
so using useEffect queries the database everytime you search (eg the dependency on searchQuery). I feel like you don't need to query the database every time, just per render. Is there a reason for this?

Reply:
oops yeah this was something we meant to revisit but we forgot. we ended up using this logic bc it parallels part of the Recipes onboarding project. i agree we should only query the database once and just filter it on search change. @nikkisuzani  check out  https://www.freecodecamp.org/news/build-a-search-filter-using-react-and-react-hooks/ for how that would work.

Reply:
also since you're first to this can you create a custom hook to fetch the data? eg `useFetchData()`